### PR TITLE
Fixed the JIRA issues and Resolved conflicts and merge

### PR DESCRIPTION
**Pull request by NipunaMarcus**

### CODE REVIEW

File: apps/portal/controllers/includes/dashboard-scripts.jag [hunk 1]
```diff
@@ -1,18 +1,21 @@
 <%
```
Comment (sinthuja) [inline apps/portal/controllers/includes/dashboard-scripts.jag]: Missing licence header.

File: apps/portal/configs/designer.json [hunk 2]
```diff
@@ -1,33 +1,36 @@
 {
```
Comment (sinthuja) [inline apps/portal/configs/designer.json]: Missing license header.

File: apps/portal/controllers/index.jag [hunk 3]
```diff
@@ -1,9 +1,12 @@
 <%
```
Comment (sinthuja) [inline apps/portal/controllers/index.jag]: Missing license header.

File: apps/portal/controllers/index.jag [hunk 4]
```diff
@@ -15,24 +18,35 @@
...
+                dashboardCountPerPortal += assetsForPortal.length;
+            }
+        });
 
-    if (dashboardCount == 1) {
-        response.sendRedirect("dashboards/" + assets[0].id);
-        return;
-    }
+        if ((dashboardCountPerPortal > 0 && dashboardCountPerUser <= 0) || dashboardCountPerUser > 1) {
+            response.sendRedirect("dashboards/");
```
Comment (sinthuja) [inline apps/portal/controllers/index.jag]: Reuse by defining it globally rather hardcoding in every places.

File: apps/portal/extensions/stores/fs/index.js [hunk 5]
```diff
@@ -20,6 +20,21 @@ var utils = require('/modules/utils.js');
         return id ? path + '/' + id : path;
     };
 
+    getDashboardsFromRegistry = function (start, count) {
+        var carbon = require('carbon');
+        var server = new carbon.server.Server();
+        var registry = new carbon.registry.Registry(server, {
+            system: true
+        });
+
```
Comment (sinthuja) [inline apps/portal/extensions/stores/fs/index.js]: Remove unnecessary blank lines.

File: apps/portal/js/create.js [hunk 6]
```diff
@@ -149,15 +149,14 @@ $(function () {
...
+            url = $("#create-dashboard-url").val() + "/" + id.val(),
             titleError = $("#title-error"),
             idError = $("#id-error");
 
         if (!$.trim(title.val()) || !$.trim(id.val())) {
             !$.trim(title.val()) ? showInlineError(title, titleError) : showInlineError(id, idError);
         } else {
-            form.attr('action', action + '/' + id.val());
+            form.attr('action', url);
             console.log("[Sending AJAX request to " + url);
```
Comment (sinthuja) [inline apps/portal/js/create.js]: Remove console.log.

File: apps/portal/js/create.js [hunk 7]
```diff
@@ -181,4 +180,8 @@ $(function () {
     menu.find('.ues-tiles-menu-toggle').click(function () {
         menu.find('.ues-tiles-menu').slideToggle();
     });
+
+    var breadcrumbs = $("#ues-breadcrumbs");
+    breadcrumbs.append("<li><a href='" + ues.utils.tenantPrefix() + "./dashboards'>Dashboards</a></li>");
```
Comment (sinthuja) [inline apps/portal/js/create.js]: Use locales without hard coding.

File: apps/portal/js/dashboard-settings.js [hunk 8]
```diff
@@ -14,15 +14,78 @@ $(function () {
...
+        properties.text = text;
+        if (ok || cancel) {
+            properties.buttons = [
+                {
+                    addClass: 'btn btn-primary', text: 'Ok', onClick: function ($noty) {
+                    $noty.close();
+                    if (ok) {
+                        ok();
+                    }
+                }
```
Comment (sinthuja) [inline apps/portal/js/dashboard-settings.js]: Formatting issue, code indentation.

File: apps/portal/js/dashboard-settings.js [hunk 9]
```diff
@@ -14,15 +14,78 @@ $(function () {
...
+                },
+                {
+                    addClass: 'btn btn-danger', text: 'Cancel', onClick: function ($noty) {
+                    $noty.close();
+                    if (cancel) {
+                        cancel();
+                    }
+                }
```
Comment (sinthuja) [inline apps/portal/js/dashboard-settings.js]: Formatting issue, code indentation.

File: apps/portal/js/dashboard-settings.js [hunk 10]
```diff
@@ -14,15 +14,78 @@ $(function () {
...
+
+    var saveDashboard = function (callback) {
         $.ajax({
             url: url,
             method: 'PUT',
             data: JSON.stringify(dashboard),
             contentType: 'application/json'
         }).success(function (data) {
+            generateMessage("Saved Successfully", null, null, "success", "bottom", 2000);
             console.log('dashboard saved successfully');
```
Comment (sinthuja) [inline apps/portal/js/dashboard-settings.js]: Remove console.log.

File: apps/portal/modules/user.js [hunk 11]
```diff
@@ -19,9 +19,18 @@ var roles;
...
         var carbon = require('carbon');
         var server = new carbon.server.Server();
-        if (!server.authenticate(username, password)) {
-            return false;
+        try {
+            if (!server.authenticate(username, password)) {
+                return false;
+            }
+        }catch (e){
+            var message = "Invalid Domain Name";
```
Comment (sinthuja) [inline apps/portal/modules/user.js]: There might be other reasons the exception could have been thrown, and those cases are swallowed. Please log actual exception when and exception occurred.

File: apps/portal/theme/templates/dashboard-settings.jag [hunk 12]
```diff
@@ -6,29 +6,26 @@
...
-            <a href="#" class="navbar-brand ues-tiles-menu-toggle"><i class="fw fw-tiles"></i></a>
+        </div>
+        <div id="ues-breadcrumbs-container">
+            <ol id="ues-breadcrumbs" class="breadcrumb ues-designer-breadcrumb">
+            </ol>
         </div>
         <div class="collapse navbar-collapse ues-context-menu-actions">
             <ul class="nav navbar-nav">
-                <li class="active"><a href="<%=tenantedUrlPrefix%>./dashboards"><%= i18n.localize("dashboards.lable")%></a></li>
+                <!--<li class="active"><a href="<%=tenantedUrlPrefix%>./dashboards"><%= i18n.localize("dashboards.lable")%></a></li>-->
```
Comment (sinthuja) [inline apps/portal/theme/templates/dashboard-settings.jag]: If this is not needed then removed it rather commenting out.

File: apps/portal/theme/templates/designer.jag [hunk 13]
```diff
@@ -15,36 +15,22 @@
...
-                <a href="#" class="navbar-brand ues-tiles-menu-toggle"><i class="fw fw-tiles"></i></a>
+            </div>
+            <div id="ues-breadcrumbs-container">
+                <ol id="ues-breadcrumbs" class="breadcrumb ues-designer-breadcrumb">
+                </ol>
             </div>
             <div class="collapse navbar-collapse ues-context-menu-actions">
                 <ul class="nav navbar-nav">
-                    <li class="active"><a href="#"><%= i18n.localize("dashboard.design") %> </a></li>
+                    <!--<li class="active"><a href="#"><%= i18n.localize("dashboard.design") %> </a></li>-->
```
Comment (sinthuja) [inline apps/portal/theme/templates/designer.jag]: Remove the line if not needed rather commenting it.

File: apps/portal/configs/designer.json (hunk 2 above)
Comment (NipunaMarcus) [inline apps/portal/configs/designer.json]: Comments are not allowed in the json files.

File: apps/portal/js/create.js (hunk 7 above)
Comment (NipunaMarcus) [inline apps/portal/js/create.js]: This code is no longer there as we change the breadcrumb functionality

File: apps/portal/controllers/includes/dashboard-scripts.jag (hunk 1 above)
Comment (NipunaMarcus) [inline apps/portal/controllers/includes/dashboard-scripts.jag]: Fixed with the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/controllers/index.jag (hunk 3 above)
Comment (NipunaMarcus) [inline apps/portal/controllers/index.jag]: Fixed with the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/controllers/index.jag (hunk 4 above)
Comment (NipunaMarcus) [inline apps/portal/controllers/index.jag]: Fixed with the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/extensions/stores/fs/index.js (hunk 5 above)
Comment (NipunaMarcus) [inline apps/portal/extensions/stores/fs/index.js]: Fixed with the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/create.js (hunk 6 above)
Comment (NipunaMarcus) [inline apps/portal/js/create.js]: Fixed with the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/dashboard-settings.js (hunk 8 above)
Comment (NipunaMarcus) [inline apps/portal/js/dashboard-settings.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/dashboard-settings.js (hunk 9 above)
Comment (NipunaMarcus) [inline apps/portal/js/dashboard-settings.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/dashboard-settings.js (hunk 10 above)
Comment (NipunaMarcus) [inline apps/portal/js/dashboard-settings.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/modules/user.js (hunk 11 above)
Comment (NipunaMarcus) [inline apps/portal/modules/user.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/theme/templates/dashboard-settings.jag (hunk 12 above)
Comment (NipunaMarcus) [inline apps/portal/theme/templates/dashboard-settings.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/theme/templates/designer.jag (hunk 13 above)
Comment (NipunaMarcus) [inline apps/portal/theme/templates/designer.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

**Pull request merged**